modules/tinymce/src/core/main/ts/api/EditorUpload.ts

```diff
@@ -173,8 +149,8 @@ const EditorUpload = (editor: Editor): EditorUpload => {
           };
         });
 
-        if (filteredResult.length > 0) {
-          changeHandler.fireIfChanged();
+        if (shouldDispatchChange && imagesToRemove.length === 0 && !Rtc.isRtc(editor)) {
```

Comment:
This logic doesn't quite look right based on the previous logic... this should fire if `shouldDispatchChange` is true and not take into account RTC. So what we probably want is to merge it into the if block below. For example:
```
if (imagesToRemove.length > 0 && !Rtc.isRtc(editor)) {
  ...
} else if (shouldDispatchChange) {
  editor.undoManager.dispatchChange();
}

Reply:
yes, I saw that the logic was not correct and I already changed it [here](https://github.com/tinymce/tinymce/pull/7840/commits/98a7038c85e19badf34c44fa4cd79d16cadf956c), but in this way have more sense I'll refactor it

Reply:
done